src/main/scala/se/nimsa/sbx/box/BoxDAO.scala

```diff
@@ -399,6 +406,14 @@ class BoxDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: ExecutionCont
     outgoingImageQuery.filter(_.outgoingTransactionId === outgoingTransactionId).result
   }
 
+  def streamPendingOutgoingImagesForOutgoingTransactionId(outgoingTransactionId: Long): DatabasePublisher[OutgoingImage] = db.stream {
+    outgoingImageQuery
+      .filter(_.outgoingTransactionId === outgoingTransactionId)
+      .filter(_.sent === false)
+      .sortBy(_.sequenceNumber.asc) // remove?
+      .result
+  }
+
```

Comment:
Uses Slick's support for streaming. This means that when sending files to another slicebox, we are streaming all the way from the database (and storage) to the receiver.